includes/notes/class-wc-stripe-upe-stripelink-note.php

```diff
@@ -0,0 +1,100 @@
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+use Automattic\WooCommerce\Admin\Notes\Note;
+use Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Stripe_UPE_StripeLink_Note
+ */
+class WC_Stripe_UPE_StripeLink_Note {
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-stripe-upe-stripelink-note';
+
+	/**
+	 * Link to enable the UPE in store.
+	 */
+	const NOTE_DOCUMENTATION_URL = 'https://woocommerce.com/document/stripe/#section-11';
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		$note_class = self::get_note_class();
+		$note       = new $note_class();
+
+		$note->set_title( __( 'Increase conversion at checkout', 'woocommerce-gateway-stripe' ) );
+		$note->set_content( __( 'Reduce cart abandonment and create a frictionless checkout experience with Link by Stripe. Link autofills your customer’s payment and shipping details so they can check out in just six seconds with the Link optimized experience. That’s 9x faster than shoppers who don’t use Link. Link increases conversion rates by over 7% for logged-in Link customers.', 'woocommerce-gateway-stripe' ) );
+
+		$note->set_type( $note_class::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_source( 'woocommerce-gateway-stripe' );
+		$note->add_action(
+			self::NOTE_NAME,
+			__( 'Enable in your store', 'woocommerce-gateway-stripe' ),
+			self::NOTE_DOCUMENTATION_URL,
+			$note_class::E_WC_ADMIN_NOTE_UNACTIONED,
+			true
+		);
+
+		return $note;
+	}
+
+	/**
+	 * Get the class type to be used for the note.
+	 *
+	 * @return string
+	 */
+	private static function get_note_class() {
+		if ( class_exists( 'Automattic\WooCommerce\Admin\Notes\Note' ) ) {
+			return Note::class;
+		} else {
+			return WC_Admin_Note::class;
+		}
+	}
+
+	/**
+	 * Init Link payment method notification
+	 *
+	 * @param WC_Stripe_UPE_Payment_Gateway $gateway
+	 *
+	 * @return void
+	 * @throws \Automattic\WooCommerce\Admin\Notes\NotesUnavailableException
+	 */
+	public static function init( WC_Stripe_UPE_Payment_Gateway $gateway ) {
+		if ( ! WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+			return;
+		}
+
+		// Check if Link payment is available.
+		$available_upe_payment_methods = $gateway->get_upe_available_payment_methods();
+
+		if ( ! in_array( WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID, $available_upe_payment_methods, true ) ) {
+			return;
+		}
+
+		// Retrieve enabled payment methods at checkout.
+		$enabled_payment_methods = $gateway->get_upe_enabled_at_checkout_payment_method_ids();
+
+		// If card payment method is not enabled, skip. If Link payment method is enabled, skip.
+		if (
+			! in_array( WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID, $enabled_payment_methods, true ) ||
+			in_array( WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID, $enabled_payment_methods, true )
+		) {
+			return;
+		}
```

Comment:
If Stripe Link is not available, will it be enabled?

We might wanna consider using the existing function `WC_Stripe_UPE_Payment_Method_Link::is_link_enabled()` to simplify it.

Reply:
`is_link_enabled ` has also other conditions added, and this will add more complexity to the code. I would keep the condition as it is now, since this way is checked also the first part of the condition (`! in_array( WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID, $enabled_payment_methods, true )`)